presto-main/src/main/java/com/facebook/presto/operator/OperatorSharedMemory.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+
+public class OperatorSharedMemory
+{
+    private int reservations = 0;
+    private Integer partitionKey;
+    private Long inMemorySizeInBytes;
+    private TaskContext taskContext;
+
+    public synchronized void setReservation(int partitionKey, OperatorContext operatorContext, long inMemorySizeInBytes)
```

Comment:
by looking at this class's interface I'm confused what `partitionKey` is. Maybe try to refactor it so it's more obvious? Rename it? As a last resort add java doc?

It doesn't seem to fit into more general concept of "shared memory". If you need it for assertions only, maybe achieve this in some other way? This seems like leakage of implementation detail of spilling.